### PR TITLE
rebased dac changes

### PR DESCRIPTION
**Pull request by alvrs**

### CODE REVIEW

File: packages/contracts-bedrock/deploy-config/hardhat.json [hunk 1]
```diff
@@ -42,5 +42,9 @@
   "l2GenesisRegolithTimeOffset": "0x0",
   "systemConfigStartBlock": 0,
   "requiredProtocolVersion": "0x0000000000000000000000000000000000000000000000000000000000000000",
-  "recommendedProtocolVersion": "0x0000000000000000000000000000000000000000000000000000000000000000"
+  "recommendedProtocolVersion": "0x0000000000000000000000000000000000000000000000000000000000000000",
+  "daChallengeWindow": 1000,
```
Comment (tchardin) [inline packages/contracts-bedrock/deploy-config/hardhat.json]: Plasma params should be optional and removed from all deploy-config files in favour of a separate plasma-devnetL1.json with the default values in there

File: packages/contracts-bedrock/scripts/DeployConfig.s.sol [hunk 2]
```diff
@@ -104,6 +108,10 @@ contract DeployConfig is Script {
         systemConfigStartBlock = stdJson.readUint(_json, "$.systemConfigStartBlock");
         requiredProtocolVersion = stdJson.readUint(_json, "$.requiredProtocolVersion");
         recommendedProtocolVersion = stdJson.readUint(_json, "$.recommendedProtocolVersion");
+        daChallengeWindow = stdJson.readUint(_json, "$.daChallengeWindow");
```
Comment (tchardin) [inline packages/contracts-bedrock/scripts/DeployConfig.s.sol]: These are assigned below

File: packages/contracts-bedrock/scripts/DeployConfig.s.sol [hunk 3]
```diff
@@ -104,6 +108,10 @@ contract DeployConfig is Script {
...
+        daResolveWindow = stdJson.readUint(_json, "$.daResolveWindow");
+        daBondSize = stdJson.readUint(_json, "$.daBondSize");
+        daResolverRefundPercentage = stdJson.readUint(_json, "$.daResolverRefundPercentage");
```
Comment (tchardin) [inline packages/contracts-bedrock/scripts/DeployConfig.s.sol]: Move this one below

**Review by tchardin — COMMENTED** [review]
Just a few changes to the deploy config

File: op-bindings/bindings/dataavailabilitychallenge.go
Comment (alvrs) [inline op-bindings/bindings/dataavailabilitychallenge.go:1]: todo: need to regenerate these

Comment (alvrs) [reply op-bindings/bindings/dataavailabilitychallenge.go:1]: done

**Pull request closed without merging**